crates/wash-lib/src/registry.rs

```diff
@@ -268,11 +280,21 @@ pub async fn parse_and_validate_artifact(artifact: Vec<u8>) -> Result<SupportedA
     }
 }
 
+/// Function that identifies whether the artifact is a component or a provider archive. Returns an
+/// error if it isn't a known type
+// NOTE: This exists because we don't care about parsing the proper world when pulling
+pub async fn identify_artifact(artifact: Vec<u8>) -> Result<ArtifactType> {
```

Comment:
Might be nice to avoid having to own it, if we could avoid it

Reply:
Yep, copy paste